File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;
    public void solve(TreeNode root, int maxVal)
    {
        if(root==null)return;
        if(root.val>=maxVal)
        {
            ans++;
            maxVal=root.val;
        }
        solve(root.left, maxVal);
        solve(root.right, maxVal);
    }
    public int goodNodes(TreeNode root) {
        int maxVal = root.val;
        solve(root, maxVal);
        return ans;
    }
}

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function(root) {
    let ans = 0;
    let solve = function(root, maxVal)
    {
        if(root==null)return;
        if(root.val>=maxVal)
        {
            maxVal=root.val;
            ans++;
        }
        solve(root.left, maxVal);
        solve(root.right, maxVal);
    }
    solve(root,root.val);
    return ans;
};